plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -687,6 +687,19 @@ public void testNestedUuid()
         assertQuery("SELECT int_t FROM test_nested_uuid WHERE row_t.uuid_t = UUID '406caec7-68b9-4778-81b2-a12ece70c8b1'", "VALUES 2");
     }
 
+    @Test
+    public void testPredicateOnNestedField()
```

Comment:
This test succeeds without the above change. Can you make the test fails without the fix?

Reply:
Fixed, thanks. I broke it while "cleaning" it up

Reply:
We should have this test in delta and hive as well. Can we put it in BaseConnectorTest (skip it for connectors which don't support insert or complex types) ?

Reply:
Hive connector test uses ORC files. We may want to add a dedicated test to Hive if there's no existing test. I confirmed the connector doesn't have this issue though.